zowe/zowe-cli
# Provide a new ConvertV1Profiles class for converting V1 profiles

**What It Does**


The new ConvertV1Profiles class enables apps to convert V1 profiles into a current Zowe client configuration file.
- Refactored logic from convert-profiles.handler and ConfigBuilder.convert into ConvertV1Profiles.convert.
- Removed ConfigBuilder.convert.
- Replaced IConfigConvertResult with IConvertV1Profiles (which contains IConvertV1ProfResult).
- Renamed class V1ProfileConversion (previously known as ProfileIO) to V1ProfileRead for clarity.
  - Marked class V1ProfileRead as @internal.

The primary function of the `convert profiles` command remains the same. The following changes were introduced:
- The command no longer prompts if you want to convert your profiles.
- The prompt asking if you want to delete your old profiles occurs before any detection of profiles and before any other information is displayed.
- Messages about all completed actions are displayed first, followed by messages for actions that failed.
- Some messages were stated more generically so that they are appropriate for both CLI and GUI apps.

The ConvertV1Profiles.convert API was written to work better for apps that are not a command line program.
- Instead of displaying messages to the console, the API places each line of message text into result object, along with formatting hints. Informational report messages are differentiated from error messages. By using the result object, all calling apps can display the same messages.
- The previous API threw exceptions in many error situations. It expected that the exception would just be displayed in a terminal. The new API catches exceptions and places error messages into the result object.

**How to Test**


The impact on end-users can be tested by running the `convert profiles` command in various scenarios. Because V1 profiles can no longer be created in V3, it can be difficult to create useful test data. The following screenshots demonstrate several known scenarios.

In a simple situation the output looks like the screenshot below. In this scenario, the V1 SCS plugin was already installed, and the `convert profiles` command uninstalled that plugin.

![image](https://github.com/zowe/zowe-cli/assets/37119146/f4706e86-5e24-4a86-af06-83b20cc55a25)

When you also ask to delete your old profiles, you are prompted to confirm the deletion. The output looks like this:

![image](https://github.com/zowe/zowe-cli/assets/37119146/5e482996-d33e-4a2b-bc2c-0b58e04b30c2)

When an error occurs, the output looks like this:

![image](https://github.com/zowe/zowe-cli/assets/37119146/5fe5582c-49b4-41fa-a191-f8269fef7614)

When the old V1 SCS plugin is configured as the CredentialManager for Zowe, the .zowe\settings\imperative.json looks like this:

```
{
  "overrides": {
    "CredentialManager": "@zowe/secure-credential-store-for-zowe-cli"
  }
}
```

When configured like this, **ALL** zowe commands will display a warning that the Credential Manager is not in the list of known credential managers. The command then attempts its operation. An example of a `zowe-files list` command shows this behavior.

![image](https://github.com/zowe/zowe-cli/assets/37119146/f8586108-01eb-4955-a2ab-052a618f92b0)

Also, when configured like this, the `plugins uninstall @zowe/secure-credential-store-for-zowe-cli` command (like all Zowe commands) displays the same CredentialManager warning. It also display a set of NPM warnings when uninstalling the V1 SCS plugin, even though it successfully uninstalls the SCS plugin.

![image](https://github.com/zowe/zowe-cli/assets/37119146/cea6027f-2a3f-4e0e-80fb-1cd090dca980)

The warnings above are not a result of the `convert profiles` command, but they do affect the `convert profiles` command because:
1. It is a zowe command and all zowe commands get the CredentialManager warnings in this scenario.
2. It automatically uninstalls the SCS plugin, so it gets the NPM warnings that occur from the `plugins uninstall` operation.

Despite the warnings in such a configuration, the `convert profiles` command converts the V1 profiles, uninstalls the SCS plugin, and restores the Zowe built-in CredentialManager.

![image](https://github.com/zowe/zowe-cli/assets/37119146/2989df94-d31f-4623-b049-cf86e7152795)

As proof that the `convert profiles` command has restored the built-in CredentialManager, when you now run the `zos-files list` command, you no longer get the CredentialManager warning.

![image](https://github.com/zowe/zowe-cli/assets/37119146/df78c8d0-5986-4af2-9d31-2129abdc659f)

**Review Checklist**
I certify that I have:


**Additional Comments**
